packages/frontend/src/components/RefreshServer/index.tsx

```diff
@@ -0,0 +1,64 @@
+import React, { ComponentProps, FC, useState } from 'react';
+import { useRefreshServer } from '../../hooks/useRefreshServer';
+import { useServerStatus } from '../../hooks/useServerStatus';
+import { useApp } from '../../providers/AppProvider';
+import { useTracking } from '../../providers/TrackingProvider';
+import { EventName } from '../../types/Events';
+import { BigButton } from '../common/BigButton';
+import {
+    LoadingSpinner,
+    RefreshButton,
+    RefreshSpinnerButton,
+} from './RefreshServerButton.style';
+import RefreshStepsModal from './RefreshStepsModal';
+
+const RefreshServerButton: FC<ComponentProps<typeof BigButton>> = (props) => {
+    const [isRefreshStepsOpen, setIsRefreshStepsOpen] = useState(false);
+    const refreshServer = useRefreshServer();
+    const status = useServerStatus();
+    const { track } = useTracking();
+    const { showToastInfo } = useApp();
+    const isLoading = status.data === 'loading';
+
+    const onClick = () => {
+        refreshServer.mutate();
+        showToastInfo({
+            title: `Sync in progress  Step 1/5: Cloning dbt project from Github`,
```

Comment:
Do we want this message to be visible until the sync ends ? @PriPatel 
If so, the title should be dynamic as the step number/label changes.

Reply:
yes, it should always show whats going on. 
**Prototype :**
 https://www.figma.com/proto/Mhuwki1Bj1akQIwPyr1gWw/Lightdash-product?node-id=3230%3A80802&scaling=min-zoom&page-id=2855%3A52273&starting-point-node-id=3207%3A83236


**Toast messages:**
https://www.notion.so/lightdash/67d3fd912f764d9eaf0f8e741d76fa00?v=805962b15e6a49958b88d1ec7dce15c4&p=2e15a707f1cf4d99b5d5f03c561eaebb

Let me know if I've misunderstood the question 😅